ci/run_shellcheck.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+set -e
+
+# Find and run shellcheck on all shell scripts.  Previously the `file` command was used
+# to identify shell commands but that checks for a #! which some of our shell code does not
+# use so go purely on filename.
+# This depends on shellcheck 0.7.2 or above so works in GitHub actions but not on el8.8
+find . \( -path ./.git -o -path ./venv -o -path ./build -o -path ./src/control/vendor \
+ -o -path ./install -o -path ./src/rdb/raft -type d  \) -prune -o -name "*.sh" -exec \
+ shellcheck -P ci --external -f gcc \{\} \+
```

Comment:
Is the argument parser doing some kind of closest-match or prefix matching or something?  https://github.com/koalaman/shellcheck/blob/master/shellcheck.1.md documents the argument as `-x, --external-sources`

Reply:
Ah, probably that.  I went through a few iterations here but this came from https://github.com/daos-stack/code_review/blob/master/shellcheck_scripts.sh#L35 which also uses `--external`.  Will expand all these options if I need to re-push.